src/daft-plan/src/logical_ops/join.rs

```diff
@@ -167,6 +169,37 @@ impl Join {
         }
     }
 
+    fn rename_join_keys(
+        left_exprs: Vec<Arc<Expr>>,
+        right_exprs: Vec<Arc<Expr>>,
+    ) -> (Vec<Arc<Expr>>, Vec<Arc<Expr>>) {
+        left_exprs
+            .into_iter()
+            .enumerate()
+            .map(|(i, left_expr)| {
+                let right_expr = &right_exprs[i];
+
+                // Rename left_expr if it's different from right_expr
+                let renamed_left_expr = if left_expr != *right_expr {
+                    let unique_id = Uuid::new_v4();
+                    left_expr.alias(format!("{}", unique_id))
+                } else {
+                    left_expr // Keep it unchanged if common
+                };
+
+                // Rename right_expr if it's different from left_expr
+                let renamed_right_expr = if *right_expr != renamed_left_expr {
```

Comment:
Why not put both the renaming in the same if statement?

Reply:
Also taken care of with the new version just committed.